### PR TITLE
CASSANDRA-17731 4.0 Clean up ScheduledExecutors, CommitLog, and MessagingService shutdown for in-JVM dtests

### PR DESCRIPTION
**Pull request by maedhroz**

### CODE REVIEW

**Comment by maedhroz** [comment]
https://app.circleci.com/pipelines/github/maedhroz/cassandra?branch=CASSANDRA-17731-4.0

**Comment by maedhroz** [comment]
Multiplexer run for `readWriteDuringBootstrapTest`:

https://app.circleci.com/pipelines/github/maedhroz/cassandra?branch=CASSANDRA-17731-4.0-multiplexer

File: test/distributed/org/apache/cassandra/distributed/test/RepairErrorsTest.java [hunk 1]
```diff
@@ -145,9 +145,7 @@ public void testRemoteStreamFailure() throws Exception
     @SuppressWarnings("Convert2MethodRef")
     private void assertNoActiveRepairSessions(IInvokableInstance instance)
     {
-        // Make sure we've cleaned up sessions and parent sessions:
-        Integer parents = instance.callOnInstance(() -> ActiveRepairService.instance.parentRepairSessionCount());
-        assertEquals(0, parents.intValue());
```
Comment (dcapwell) [inline test/distributed/org/apache/cassandra/distributed/test/RepairErrorsTest.java:150]: why?

Comment (dcapwell) [reply test/distributed/org/apache/cassandra/distributed/test/RepairErrorsTest.java:150]:
```On the RepairErrorsTest thing, that was causing testRemoteSyncFailure to flake every now and then...looked, and it seems that's updated async, so just related to look at the local sessions.
(nobody had reported a Jira for that one, but you can see it in Butler)```

**Review by dcapwell — APPROVED** [review]

**Comment by maedhroz** [comment]
Committed in https://github.com/apache/cassandra/commit/d51f90201b2d127166612384a75b4a49da9776d1

**Pull request closed without merging**